### PR TITLE
File defects

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: spot_data_responses.go [hunk 1]
```diff
@@ -181,5 +211,50 @@ func (r *SingleMarketStatistics) Parse(raw_response *http.Response) (*SingleMark
 		return nil, err
 	}
 	defer raw_response.Body.Close()
-	return r, err
+	return r, nil
+}
+
+func (r *AllMarketStatistics) Parse(raw_response *http.Response) (*AllMarketStatistics, error) {
```
Comment (arshamalh) [inline spot_data_responses.go:217]: Do not write all methods of "different types" under each other,
Instead, Write methods under their own type.

Do this:
```go
type sth1 struct{}
func (s sth1) Parse1() {}
func (s sth1) Parse2() {}

type sth2 struct{}
func (s sth2) Parse1() {}
func (s sth2) Parse2() {}
```

File: spot_data_responses.go [hunk 2]
```diff
@@ -104,6 +104,36 @@ type SingleMarketStatistics struct {
...
+// AllMarketStatistics
+type AllMarketStatistics struct {
+	GeneralResponse
+	Data TickerData
+}
+
+// CurrencyRate
+type CurrencyRate struct {
+	GeneralResponse
+	Data TickerData
```
Comment (arshamalh) [inline spot_data_responses.go]: Define a custom type instead of wrongly using TickerData everywhere,
we've spoken about it in detail earlier.

**Comment by arshamalh** [comment]
**Do not open different pull requests for one purpose**
@arghvn

**Comment by ghost** [comment]
fix  AllMarketStatistics again

**Pull request closed without merging**